testdata/if.zsh

```diff
@@ -0,0 +1,26 @@
+# Test "if" and related constructs in various way.
+
+if [ $foo ]; then
+	:
+elif [[ $foo ]]; then
+	:
+elif (( $foo )); then
+	:
+fi
+
+if $(grep x y); do
+	:
+elif [[ $(grep x y) -gt 42 ]]; then
+	:
+elif (( $(grep x y) > 42 )); then
+	:
+fi
+
+[[ $(grep x y) -gt 42 ]] && print x
+(( $(grep x y) > 42 ))   && print x
+
+# Unlike [/[[, (( doesn't need spaces.
+[[ x = y ]]&&((1>0))&& print x
+
+# TODO: maybe show error for these invalid constructs (missing spaces)?
+[[x=x]] && [x=x]
```

Comment:
These aren't invalid constructs; they're perfectly valid glob patterns.  The first looks for a file named `[]` or `=]` or `x]`.  The second one looks for a file named `x` or `=`.

Nevertheless, I agree that it might make sense to highlight these constructs as "this is probably wrong" (compare https://github.com/zsh-users/zsh-syntax-highlighting/issues/695 and the issue linked from the first post there).